app/src/renderer/vuex/modules/node.js

```diff
@@ -67,6 +72,25 @@ export default function ({ node }) {
         commit('notifyError', {title: 'Critical Error', body: `Couldn't initialize blockchain connector`})
         return false
       }
+    },
+    pollRPCConnection ({state, commit, dispatch}, timeout = 3000) {
+      if (state.nodeTimeout) return
+
+      state.nodeTimeout = setTimeout(() => {
+        // clear timeout doesn't work
+        if (state.nodeTimeout) {
+          state.nodeTimeout = null
+          dispatch('reconnect')
+        }
+      }, timeout)
+      node.rpc.status((err, res) => {
+        if (!err) {
+          state.nodeTimeout = null
```

Comment:
Should this call `clearTimeout`?

Reply:
Oh, I see the comment now. Why doesn't it work? I didn't think nulling the return value would work 🤔

Reply:
Nvm, missed the `if (state.nodeTimeout)`. :+1: